File: 047. Permutations II/Java/Solution.java
public class Solution {
    public List<List<Integer>> permuteUnique(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        result.add(new ArrayList<>());
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            List<List<Integer>> currentResult = new ArrayList<>();
            for (List<Integer> item : result) {
                for (int j = 0; j <= i; j++) {
                    if (j > 0 && num == item.get(j - 1)) {
                        break;
                    }
                    List<Integer> temp = new ArrayList<>(item);
                    temp.add(j, num);
                    currentResult.add(temp);
                }
            }
            result = currentResult;
        }
        return result;
    }
}

File: 047. Permutations II/C++/Solution.cpp
class Solution {
public:
    vector<vector<int>> permuteUnique(vector<int> &nums) {
        vector<vector<int>> result;
        result.push_back(vector<int>());
        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            vector<vector<int>> currentResult;
            for (const auto &item : result) {
                for (int j = 0; j <= i; j++) {
                    if (j > 0 && num == item[j - 1]) {
                        break;
                    }
                    vector<int> temp = item;
                    temp.insert(temp.begin() + j, num);
                    currentResult.push_back(temp);
                }
            }
            result = currentResult;
        }
        return result;
    }
};